src/hipoc/hipoc_program.cpp

```diff
@@ -62,10 +62,6 @@ MIOPEN_DECLARE_ENV_VAR(MIOPEN_DEBUG_USE_HIPRTC)
 
 #define MIOPEN_WORKAROUND_ISSUE_1359 1
 
-#if MIOPEN_USE_COMGR
-#define MIOPEN_WORKAROUND_ROCM_COMPILER_SUPPORT_ISSUE_27 1
```

Comment:
We need to know the HIP version from which the defect was fixed and define this macro to 0 from that version. HIP version can be taken from the hip-clang mainline where the defect is fixed (it should be reported in SWDEV).
```C++
#define MIOPEN_WORKAROUND_ROCM_COMPILER_SUPPORT_ISSUE_27 (HIP_PACKAGE_VERSION_FLAT < xyyyzzzzzzULL)
```
This should resolve https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1977/#issuecomment-1426844139

See #1685 for more info about W/A lifecycle, thanks.

Reply:
Will do